### ISSUE TITLE
A mechanism for renaming metrics that isn't env vars

### ISSUE THREAD

**Issue by rthomas** [Ptrskay3/axum-prometheus#10]
The current mechanism for renaming metrics (env vars) does not work well when you have multiple projects using `axum-prometheus` in a cargo workspace.

Specifically, from https://doc.rust-lang.org/cargo/reference/config.html:

> At present, when being invoked from a workspace, Cargo does not read config files from crates within the workspace.

It would be great to provide a mechanism to supply a prefix for these metrics.

Using `env::set_var` or `cargo:rustc-env=` from `build.rs` files does not appear to work either, as a workaround.

**Comment by Ptrskay3** [comment]
I guess you can just do 
```bash
export AXUM_HTTP_REQUESTS_TOTAL=something [...others] cargo build
```
when there's only one `axum_prometheus` instance across your workspace; it obviously doesn't work for multiple. Since cargo won't read config from crates, the only real solution is a way specifically from inside the code to manipulate the name of the metrics (or name prefix rather). 

Is that what you're looking for?

**Comment by rthomas** [comment]
We have multiple consumers of this crate within our workspace, that are all
built at once, so unfortunately this doesn’t work for our use case.

**Comment by leeh-peter** [comment]
> the only real solution is a way specifically from inside the code to manipulate the name of the metrics (or name prefix rather). 
Is that what you're looking for?

I'm asking this part specifically. :smile: 

I meant something like a new API in the library, for instance:
```rust
PrometheusMetricBuilder::prefix("my_app").build_pair();
```

Is by any chance the repository public for your use case?

**Comment by Ptrskay3** [comment]
> the only real solution is a way specifically from inside the code to manipulate the name of the metrics (or name prefix rather). 
Is that what you're looking for?

I'm asking this part specifically. :smile: 

I meant something like a new API in the library, for instance:
```rust
PrometheusMetricBuilder::prefix("my_app").build_pair();
```

Is by any chance the repository public for your use case?

**Comment by rthomas** [comment]
Sorry :) an API like that would be perfect for our needs. For more context
we have a workspace with all our rust services in it, as well as a bunch of
common code - which is where axum-prometheus was going to slot in for our
metrics handling. Unfortunately this isn’t public.

I can look at raising a PR if you’re open to this change, but that won’t
happen until Monday (I’m in AU and it’s Friday evening already).

**Comment by Ptrskay3** [comment]
The change is reasonable, I'd greatly appreciate a PR. I'll be quite busy in the next period and I'm not sure I'd find the time to implement this fully myself, however I can definitely help with moving forward with it and review.

**Comment by rthomas** [comment]
Sounds great, I’ll jump on it on Monday!

**Comment by Ptrskay3** [comment]
Just published 0.3.1, and it includes your change.

**Comment by rthomas** [comment]
Awesome, thanks!
